### PR TITLE
Concurrent execution of the event handlers

### PR DESCRIPTION
**Pull request by nkuba**
In this PR we allow concurrent execution of event handlers for a given subscription. Currently, when a new event comes it blocks the execution of handlers for any subsequent events.

### CODE REVIEW

**Comment by pdyraga** [comment]
I am not sure about this change but would love to hear your arguments and opinion of @Shadowfiend.

I think it makes sense to do not trigger chain event handlers asynchronously. Chain processes transactions one after another and I can imagine we (or anyone else using this library) may want to process events in order as they appeared on-chain. 

Although, in my opinion, we shouldn't fire goroutines for event handler in the code of generated contracts I do recognize the problem and I think we should fire those goroutines inside the callback function, just like we do in `keep-core`. One example [here](https://github.com/keep-network/keep-core/blob/master/pkg/beacon/beacon.go#L75), another example [here](https://github.com/keep-network/keep-core/blob/master/pkg/beacon/beacon.go#L114).

Basically, the client code should decide if it cares about the order of events or not and trigger time-consuming handlers asynchronously if that's needed.

**Comment by nkuba** [comment]
After pushing this PR I had similar second thoughts. The longer I test the client the more I doubt we need this.

**Comment by pdyraga** [comment]
It's perfectly fine to fire callbacks in goroutines if needed, I just don't think we should do it by default. Do we want to close this PR without merging?
